Documentation/building/dev-workflow.md

```diff
@@ -0,0 +1,46 @@
+Dev Workflow
+===============
+
+The dev workflow describes the development process to follow when working in our repo. 
+It is divided into specific tasks that are fast, transparent and easy to understand.
+
+This provides flexibility as with discrete tasks people can iterate on a specific one without needing to run the full workflow, other processes can reuse the tasks as needed and anyone can put together the tasks as wanted to create a customized workflow.
+
+## Process 
+![Dev Workflow process](../images/dev-workflow.png)
+
+All the tasks (except enlist, code, submit) will have scripts located at the root of the repo, and will have usage help to better understand what they provide.
```

Comment:
I agree enlist is a little TFS specific, however clone is also git specific. Perhaps we should call it "setup" or something along those lines. Also while there isn't aren't any scripts associated with it there are instructions which we should list here in the documentation so folks know what they need to do to get going. This kind of connects to my other comment that we need to weave this with the actually build instructions.